bootloaders/riotboot/README.md

```diff
@@ -112,3 +104,47 @@ flash` if the `riotboot` feature is used.
 ## Testing riotboot
 
 See [tests/riotboot/README.md](https://github.com/RIOT-OS/RIOT/blob/master/tests/riotboot/README.md).
+
+# Quick riotboot porting guide
+
+Your board/cpu is not supported yet? Try to port riotboot!
+
+## Porting to a board based on am Arm Cortex-M0+/3/4/7 MCU
+
+Extending riotboot to support another board with a Cortex-M0+/3/4/7
+microcontroller is rather straightforward. You need to:
+
+- Provide the variables `ROM_START_ADDR` and `ROM_LEN` as well as
+`RAM_LEN` and `RAM_START_ADDR`.
+- Adapt the linker scripts for your cpu to pass the test
+[tests/cortexm_common_ldscript](../../tests/cortexm_common_ldscript). This test
+ensures that the linker script supports building firmware's with a rom offset
+and specific sized firmware's.
+- Make the startup code `board_init()`, `cpu_init` idempotent, i.e. whether you
+execute it once, or twice, it works all the same. The global define `RIOTBOOT`
+can be useful here. (e.g. [cpu/efm32/cpu.c](../../cpu/efm32/cpu.c))
+- Make sure (and adapt if needed) that the flasher script can:
+    - flash a `.bin`
+    - flash at a specified offset
+    - flash without performing mass erase, only erase flash sections to be written
+  e.g.: [makefiles/tools/edbg.inc.mk](../../makefiles/tools/edbg.inc.mk)
+- Declare `FEATURES_PROVIDED += riotboot` for the target `BOARD`/`CPU`.
+- Make sure that `RIOTBOOT_LEN` size is such that the remainder of the flash
+can be divided by the number slots while staying `FLASHPAGE_SIZE` aligned.
+e.g: [cpu/nrf52/Makefile.include](../../cpu/nrf52/Makefile.include)
+- Check other specific `BOARD`/`CPU` flash alignment requirements (e.g.:
+kinetis vector table must be naturally aligned to the power of two, see
+[cpu/kinetis/Makefile.include](../../cpu/kinetis/Makefile.include))
+
+## Porting to a board based on other types of MCUs
+
+For other MCU architectures the following extra requirements must be fulfilled:
+
+- Provide the functions defined in the header [sys/include/riotboot/slot.h](../../sys/include/riotboot/slot.h), in particular `riotboot_slot_jump(unsigned slot)`
+which will allow jumping from the bootloader to another slot/application.
+- Adapt the linker script so that it supports building firmware's with a ro
+offset and specific sized firmware's. To get an idea look at [cpu/cortexm_common/Makefile.include](../../cpu/cortexm_common/Makefile.include)
```

Comment:
```suggestion
- Adapt the linker script so that it supports building firmwares with a rom
offset and specific sized firmware's. To get an idea look at [cpu/cortexm_common/Makefile.include](../../cpu/cortexm_common/Makefile.include)
```

I'm not I understand what "specific sized firmware's" means.

Reply:
You need to be able to specify in the linker script the slot sizes

Reply:
Reworded